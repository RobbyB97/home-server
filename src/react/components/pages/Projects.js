import React from 'react'
import {connect} from 'react-redux'

import PrevPage from '../common/PrevPage'
import NextPage from '../common/NextPage'
import Project from '../content/Project'

import {
    project_ytIframe, 
    project_thisWebsite, 
    project_webDevBootcamp
} from '../../data/content/Project'


export class Projects extends React.Component {

    render() {
        return (
            <div id="projects">
                Projects component

                <Project {...project_ytIframe}/>
                <Project {...project_thisWebsite}/>
                <Project {...project_webDevBootcamp}/>
                
                <PrevPage
                    text="Experience"
                    route="/experience/"
                    float={true}
                />

                <NextPage 
                    text="Resume"
                    route="/resume/"
                    float={true}
                />
            </div>
        )
    }
}


const mapStateToProps = (state) => {
    return {
      nav: state.navigation.nav,
      title: state.title.title,
    }
}

const mapDispatchToProps = (dispatch) => {
    return {

    }
}

export default connect(mapStateToProps, mapDispatchToProps)(Projects)