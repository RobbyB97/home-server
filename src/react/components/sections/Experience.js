import React from 'react';


export class Experience extends React.Component {
    render() {
        return (
            <section id="Experience" className="experience">
                
            </section>
        );
    }
}


export default Experience;