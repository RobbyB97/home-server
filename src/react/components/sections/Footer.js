import React from 'react';

import Divider from '../Divider'


export class Footer extends React.Component {
    render() {
        return (
            <footer id="Footer" className="footer">
                <Divider section="footer__divider"/>
                
                <div className="footer__content">
                    <div className="footer__links">
                        
                    </div>

                    <p className="footer__copyright">
                        Â© Robert Bergers 2017-2020
                    </p>
                </div>
            </footer>
        );
    }
}


export default Footer;