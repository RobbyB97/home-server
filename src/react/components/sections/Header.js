import React from 'react';

import DesktopNavigation from '../ui/DesktopNavigation';
import MobileNavigation from '../ui/MobileNavigation';
import HomeButton from '../ui/HomeButton';



export class Header extends React.Component {
    render() {
        return (
            <header id="Header" className="header">
                <HomeButton internal />
                <DesktopNavigation />
                <MobileNavigation />
            </header>
        );
    }
}


export default Header;