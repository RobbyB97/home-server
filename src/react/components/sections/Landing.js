import React from 'react';
import Particles from 'react-particles-js';
import ReactModal from 'react-modal'


export class Landing extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            emailModal: false
        };
    }

    /* Controls email modal */
    toggleEmail() {
        this.setState({
            emailModal: !this.state.emailModal
        });
    }
    render() {
        return (
            <section id="Landing" className="landing">

                {/* Background color blocks */}
                <div className="landing__background">
                    <div className="one">
                        <div className="texture">
                            <div></div>
                        </div>
                    </div>
                    <div className="two">
                        <div className="texture">
                            <div></div>
                        </div>
                    </div>
                    <div className="three">
                        
                    </div>
                    <div className="four">
                        <Particles 
                            width="100vw"
                            height="100vh"
                            className="landing__particles"
                            params={{
                                particles: {
                                    number: {
                                        value: 110,
                                        density: {
                                            enable: true,
                                            value_area: 700
                                        }
                                    },
                                    line_linked: {
                                        enable: true,
                                        distance: 200,
                                        color: '#f8f8f8',
                                        opacity: 0.5,
                                        width: 1
                                    },
                                    color: {
                                        value: '#f8f8f8'
                                    }
                                }
                            }}
                        />
                    </div>
                    <div className="five"></div>
                </div>
                    
                {/* Tagline */}
                <section className="landing__tagline">
                    <h2>
                        <span className="left">Robby</span> <span className="right">Bergers</span>
                    </h2>
                    <p className="one">
                        Designer
                    </p>
                    <p className="two">
                        Developer
                    </p>
                </section>

                {/* Menu buttons */}
                <section className="landing__menu">
                    <a className="landing__view" href="#About">
                        View my work
                    </a>
                    
                    <a className="landing__email"
                        onClick={() => {
                            this.toggleEmail();
                        }}
                    >
                        Get in touch
                    </a>
                </section>

                {/* Email modal */}
                <section className={this.state.emailModal ? "landing__emailModal active":"landing__emailModal"}>
                    <p>Email me @ <a href="mailto:bergersr97@gmail.com">bergersr97@gmail.com</a></p>
                </section>

                {/* Link icons */}
                <section className="landing__contact">
                    <a className="landing__github" href="https://www.github.com/robbyb97">
                        <div></div>
                    </a>
                    
                    <a className="landing__linkedin" href="https://www.linkedin.com/in/robbyb97/">
                        <div></div>
                    </a>

                    <a className="landing__upwork" href="https://www.upwork.com/o/profiles/users/~01f696e65011a54866/">
                        <div></div>
                    </a>

                    <a className="landing__twitter" href="https://twitter.com/RobbyBergers">
                        <div></div>
                    </a>
                </section>
            </section>
        );
    }
}


export default Landing;