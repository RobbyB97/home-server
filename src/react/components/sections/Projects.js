import React from 'react';


export class Projects extends React.Component {
    render() {
        return (
            <section id="Projects" className="projects">
                <div className="projects__shape one"></div>
                <div className="projects__shape two"></div>
                Projects component
            </section>
        );
    }
}


export default Projects;