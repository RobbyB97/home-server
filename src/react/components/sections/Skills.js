import React from 'react';

import SkillSection from '../SkillSection';


export class Skills extends React.Component {
    render() {
        const sections = [
            {
                title: "Front end",
                skills: [
                    "React", "SASS", "HTML5", "CSS3", "JS (ES6)"
                ]
            }, {
                title: "Workflow",
                skills: [
                    "Git", "NPM", "Gulp", "Webpack", "Jest", 
                    "Firebase", "Heroku"
                ]
            }, {
                title: "Back end",
                skills: [
                    "Node.js", "BASH", "Linux/Ubuntu server",
                    "MongoDB", "Oracle SQL"
                ]
            }
        ];

        return (
            <section id="Skills" className="skills">
                <header className="skills__header">
                    <h2>My skillset</h2>
                </header>

                {sections.map((section) => (
                    <SkillSection 
                        {...section}
                        key={section.title}
                    />
                ))}
            </section>
        );
    }
}


export default Skills;