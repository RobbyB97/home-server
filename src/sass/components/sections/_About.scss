#About {}

.about__content {
    display: inline-block;
    font-family: $montserrat;
    text-align: center;
    width: 100%;
    padding-bottom: 20px;

    &.one {
        background-color: lighten($black, 15%);
    }
}

.about__bioTitle {
    font-size: 1.4em;
    font-weight: bold;
}

.about__portrait {
    margin: 0 auto;
    img {
        width: 300px;
        height: auto;
        border-radius: 50%;

        @media #{$bp-tablet} {
            width: 400px;
        }

        @media #{$bp-desktop} {
            width: 500px;
        }
    }
}

.about__text {
    width: 100%;
    font-size: 1.1em;
    
    p {
        width: 280px;
        margin: 0 auto;
    }
}

.about__text.one {
    position: absolute;

    p {
        position: relative;
        bottom: 60px;

        span.one {
            padding: 1px;
            background-color: $black;
        }

        span.two {
            padding: 1px;
            font-size: 1.2em;
            background-color: $red;
        }
    }
}

.about__text.two {
    margin-top: 30px;

    @media #{$bp-phone} {
        margin-top: 24px;
    }

}