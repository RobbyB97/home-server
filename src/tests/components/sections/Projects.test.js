import React from 'react';
import { shallow } from 'enzyme';

import { Projects } from '../../../react/components/sections/Projects';


test('Render Projects', () => {
    const wrapper = shallow(<Projects />);
    expect(wrapper).toMatchSnapshot();
});