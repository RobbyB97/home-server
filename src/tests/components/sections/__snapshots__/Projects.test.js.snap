// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Projects 1`] = `
<section
  className="projects"
  id="Projects"
>
  <div
    className="projects__shape one"
  />
  <div
    className="projects__shape two"
  />
  Projects component
</section>
`;
